src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -138,19 +139,35 @@ public MesosApi(String masterUrl, String user, String frameworkName)
   }
 
   /**
-   * Enqueue spec for a Jenkins agent that will eventually come online.
+   * Enqueue spec for a Jenkins event, passing a non-null existing podId will trigger a kill for
+   * that pod
    *
    * @return a {@link MesosSlave} once it's queued for running.
    */
-  public CompletionStage<MesosSlave> enqueueAgent() throws IOException, FormException {
-    PodSpec spec = buildMesosAgentTask(0.1, 32);
+  public CompletionStage<QueueOfferResult> killAgent(String id) throws IOException, FormException {
+    PodSpec spec = getKillSpec(id);
+    SpecUpdated update = new PodSpecUpdated(spec.id(), Option.apply(spec));
+    specMap.put(spec.id(), spec);
```

Comment:
We have to check that the agent exists.

Reply:
is it sufficient to check that we have a specId in the map for that pod?

Reply:
As I imagine it we get the `MesosSlave` and have it construct the `PodSpecUpdated` call. I've talked to @timcharper and we are going to simplify the calls.

Reply:
updated to get the mesosSlave